3/apollo.js

```diff
@@ -20,85 +18,156 @@ const typeDefs = gql`
+			MATCH (t:Todo)-[:IS_ASSIGNED_TO]->(p:Person)
+			RETURN t, p
+			ORDER BY p.name
+			SKIP $s
+			LIMIT $l
+			`
+			if(args.desc == true){
+				query = `
+				MATCH (t:Todo)-[:IS_ASSIGNED_TO]->(p:Person)
+				RETURN t, p
+				ORDER BY p.name DESC
+				SKIP $s
+				LIMIT $l
+				`
+			}
+			result = await session.run(query,
+			{
+				s: args.offset,
+				l: args.first
+			})			
 		}
-		return result;
+
+		session.close()
+		return result.records.map((record) => {
+			let response = {...record.get('t').properties, assignee: record.get('p').properties}
+			return response;
+		});
 	},
   },
   Mutation: {
-	  login: (parent, args, context, info) =>{
-		  return jwt.sign({userName: args.userName}, secret, {expiresIn: "1 day"});
+	  login: async(parent, args, context, info) =>{
+		  const session = context.driver.session()
+		  const result = await session.run(`
+			  MATCH (p:Person {name: $name, password: $password})
+			  RETURN p
+			  `,
+		  {
+			name: args.userName,
+			password: args.pwd
+		  }
+		  )
+		  const record = result.records[0];
+		  if(record){
+			  return jwt.sign({userName: args.userName, }, secret, {expiresIn: "1 day"});
+		  }else{
+			  return ""
+		  }
 	  },
-	  createTodo: (parent, args, context, info) =>{
-		  todos.push({
-			  id: getRandomId(),
-			  message: args.message,
-			  assignee: assignees.find((assignee)=>{ return assignee.id == args.assignee})
-		  });
-		  return todos[todos.length -1];
+	  createTodo: async (parent, args, context, info) => {
+		  const session = context.driver.session()
+		  const result = await session.run(`
+		  MATCH (p:Person) WHERE p.id = $assignee
+		  CREATE (t:Todo {id: $id, message: $message})
+		  MERGE (t)-[:IS_ASSIGNED_TO]->(p)
+		  RETURN t, p
+		  `,
+		  {
+			id: getRandomId(),
+			message: args.message,
+			assignee: args.assignee
+		  }
+		  )
+		  session.close();
+		  const record = result.records[0];
+		  return {...record.get('t').properties, assignee: record.get('p').properties};
 	  },
 	  deleteTodo: (parent, args, context, info) =>{
-		return todos.splice(todos.findIndex((todo)=>{return todo.id == args.id}),1)[0];  
+		return neo4jgraphql(parent, args, context, info)
 	  },
 	  updateTodo: (parent, args, context, info) =>{
-		let todo = todos.find((todo)=>{return todo.id == args.id});
-		todo.message = args.message;
-		return todo;
+		  return neo4jgraphql(parent, args, context, info)
 	  },
-	  createAssignee: (parent, args, context, info) =>{
-		  assignees.push({
+	  createAssignee: async (parent, args, context, info) => {
+		  const session = context.driver.session()
+		  const result = await session.run(`
+		  CREATE (p:Person:Assignee {id: $id, name: $name, password: $password})
+		  RETURN p
+		  `,
+		  {
 			  id: getRandomId(),
-			  name: args.name
-		  });
-		  return assignees[assignees.length -1];
+			  name: args.name,
+			  password: args.password
+		  }
+		  )
+		  session.close();
+		  return result.records[0].get('p').properties;
```

Comment:
If, for any reason, your records is empty, this will throw a null pointer error. Possible scenarios are a unique constraint on the `id` of a node with label `:Person`.